zkevm-circuits/src/state_circuit/lexicographic_ordering.rs

```diff
@@ -302,25 +302,27 @@ impl<F: Field> Queries<F> {
             .chain(self.rw_counter_limbs.iter().rev())
             .cloned()
             .collect();
-        // most significant byte of id should be 0, so safe to overwrite it with packed
-        // tags.
-        limbs[0] = limbs[0].clone() + self.packed_tags() * (1u64 << 8).expr();
+        // most significant 9 bits of id are assumed be 0, so we can overwrite them with
+        // packed tags.
+        limbs[0] = limbs[0].clone() + self.packed_tags() * (1u64 << 7).expr();
         limbs
     }
 }
 
 fn rw_to_be_limbs(row: &Rw) -> Vec<u16> {
+    let mut id = row.id().unwrap_or_default() as u32;
+    assert_eq!(id.to_be_bytes().len(), 4);
+    // check that the most significant 9 bits of id are 0, and pack tag and then
+    // field_tag into them.
```

Comment:
Perhaps add some short explanation why the most significant 9 bits are 0?

Reply:
updated in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/523/commits/f8314b0ebe9246744344c56f02c89ab327ac83d3 thanks !